ca/server.go

```diff
@@ -525,7 +527,8 @@ func (s *Server) UpdateRootCA(ctx context.Context, cluster *api.Cluster) {
 
 	// If the cluster has a RootCA, let's try to update our SecurityConfig to reflect the latest values
 	rCA := cluster.RootCA
-	if len(rCA.CACert) != 0 && len(rCA.CAKey) != 0 {
+	externalCACert := rCA.CACert
```

Comment:
I added a field to `api.ExternalCA` specifying which root certificate should be associated with the external URL so we can make sure that only the right external URLs are being added.  So during root rotation, only the external URLs with the right root certificate get added to the external URL.

This is the URL to filter for.  Although I guess if we are explicitly not supporting `external->external` rotation, this change is probably not necessary.